src/nanotron/generation/decode.py

```diff
@@ -190,6 +190,16 @@ def decode_text(
 
     p2p = model.p2p
 
+    if generation_config and generation_config.n_samples:
+        assert isinstance(generation_config.n_samples, int) and generation_config.n_samples > 0
```

Comment:
this arg is alrdy validated as an int. Remove this line

Reply:
resolved